test/scripts/hexo/validate_config.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const { spy } = require('sinon');
+
+describe('Validate config', () => {
+  const Hexo = require('../../../lib/hexo');
+  const hexo = new Hexo();
+  const validateConfig = require('../../../lib/hexo/validate_config');
+  const defaultConfig = require('../../../lib/hexo/default_config');
+  let logSpy;
+
+  beforeEach(() => {
+    logSpy = spy();
+    hexo.config = JSON.parse(JSON.stringify(defaultConfig));
+    hexo.log.warn = logSpy;
+    hexo.log.info = spy();
+  });
+
+  it('config.url - undefined', () => {
+    delete hexo.config.url;
+
+    try {
+      validateConfig(hexo);
+    } catch (e) {
```

Comment:
for more accurate test, suggest to have `should.fail()` right after `validateConfig()`.

Reply:
Added in https://github.com/hexojs/hexo/pull/4381/commits/c2d34280969fc774e62c3c8894732be195f5912b